### ISSUE TITLE
API Testing using Bruno

### ISSUE THREAD

**Issue by vasujain275** [fossunited/fossunited#697]
### Issue description

I created this issue to document the entire API testing integration process in one place.

@rahulporuri, as discussed, I’ve explored Bruno for our API testing needs. We can set up all API tests in Bruno and then run them using the [Bruno CLI](https://docs.usebruno.com/bru-cli/overview) either in CI with GitHub Actions, on local developer setups, or both.

I’ll start with the endpoints in the [utils file](https://github.com/fossunited/fossunited/blob/develop/fossunited/fossunited/utils.py) and then move on to the endpoints in the `api` directory.

**Comment by rahulporuri** [comment]
Thanks for creating the tracker issue @vasujain275 . Looking forward to the API test PRs

**Comment by vasujain275** [comment]
Sorry for the inactivity here, I have my end of the semester evaluations and exam going on till 18th of this month. I wiill resume the work on this issue after that.

**Comment by rahulporuri** [comment]
> Sorry for the inactivity here, I have my end of the semester evaluations and exam going on till 18th of this month. I wiill resume the work on this issue after that.

There is no need to apologize @vasujain275 ! Good luck with your exams. We'll still have plenty of work after you're done with the end sems :D

**Comment by vasujain275** [comment]
@rahulporuri @harshtandiya 

I was working on the api testing issue today and want to ask a doubt. I was writing a bruno test for `check_if_chapter_member` [endpoint](https://github.com/fossunited/fossunited/blob/develop/fossunited/api/chapter.py), but need a chapter id and user email to as params to verify or fail the test. How do I ensure that whatever sample user I pick will be a chapter member and vice versa for other test. I think we should create dummy users in our db for api testing so that we can know for sure that this user will be in chapter always and this user will be not. Like for the `get_foss_profile` endpoint I used rahul's email to write the test.

**Comment by rahulporuri** [comment]
@vasujain275 can we switch to local testing instead of testing against production? then we can add/update the necessary chapters/members (manually or via scripts) as needed for the tests.

i don't think we want to go towards having testing accounts/dummy users in production.

**Comment by vasujain275** [comment]
@rahulporuri Yes switching to localhost for testing would make this alot easier and flexible, we can then integreate it with our ci cd pipelines easliy too.

**Comment by idlip** [comment]
Okay... Bruno seems interesting as I attend their workshop in RVCE (indiafoss 25 pre events)

Actually yesterday I read an article on cURL: https://justuse.org/curl/
(cURL supremacy is real!)

I'll try it out some time and document to docs as well

I was using emacs verb package (https://github.com/federicotdn/verb) to do API testing
But my lazya$$ way to perform is `bench console` and console.log (lol)
